static/src/javascripts/projects/commercial/modules/brazeBanner.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+
+import type {Banner} from 'common/modules/ui/bannerPicker';
+import config from 'lib/config';
+import reportError from 'lib/report-error';
+import {onConsentChange} from '@guardian/consent-management-platform';
+import {mountDynamic} from "@guardian/automat-modules";
+
+import {getUserFromApi} from '../../common/modules/identity/api';
+import {isDigitalSubscriber} from "../../common/modules/commercial/user-features";
+
+const brazeSwitch = config.get('switches.brazeSwitch');
+const apiKey = config.get('page.brazeApiKey');
+const brazeVendorId = '5ed8c49c4b8ce4571c7ad801';
+
+const getBrazeUuid = (): Promise<?string> =>
+    new Promise((resolve) => {
+        getUserFromApi(user => {
+            if (user && user.privateFields && user.privateFields.brazeUuid) {
+                resolve(user.privateFields.brazeUuid);
+            } else {
+                resolve();
+            }
+        })
+    });
+
+const hasRequiredConsents = (): Promise<boolean> =>
+    new Promise((resolve) => {
+        onConsentChange(({ tcfv2, ccpa }) => {
+            const consentGivenUnderCcpa = ccpa && !ccpa.doNotSell;
+            const consentGivenUnderTcfv2 = tcfv2 && tcfv2.vendorConsents[brazeVendorId];
+
+            resolve(!!(consentGivenUnderCcpa || consentGivenUnderTcfv2));
+        })
+    });
+
+let messageConfig;
```

Comment:
Is it worth explicit typings here as they are uninitialised here, so it's clearer for the reader and safer codewise?

Reply:
I added some Flow type definitions for these. The `MessageConfig` extras format can be made more concrete in future as we decide exactly what information we'll get passed by Braze (extras are the mechanism we're using the thread information from Braze to the platform).